lib/mirage.mli

```diff
@@ -376,18 +379,20 @@ val direct_stackv4 :
 val socket_stackv4 :
   ?group:string -> Ipaddr.V4.t list -> stackv4 impl
 
+type arp_sig = ?clock:mclock impl -> ?time:time impl -> ethernet impl -> arpv4 impl
+
 (** Build a stackv4 by looking up configuration information via QubesDB,
  *  building an ipv4, then building a stack on top of that. *)
-val qubes_ipv4_stack : ?group:string -> ?qubesdb:qubesdb impl -> network impl -> stackv4 impl
+val qubes_ipv4_stack : ?group:string -> ?qubesdb:qubesdb impl -> ?arp:arp_sig -> network impl -> stackv4 impl
```

Comment:
you can let people build their own `arp` implementation (using the clock they want), I don't think we need to thread that through all the arguments.